src/toolkit/matl_buy_policy.h

```diff
@@ -128,15 +141,18 @@ class MatlBuyPolicy : public Trader {
   /// is idempotent.
   void Stop();
 
+  double sample_fraction_ = 1.0;
+
   /// the total amount requested
   inline double TotalQty() const {
-    return std::min(throughput_,
-                    fill_to_ * buf_->capacity() - buf_->quantity());
+    return std::min((throughput_* sample_fraction_),
+                    ((fill_to_ * buf_->capacity()) - buf_->quantity()) * sample_fraction_);
```

Comment:
Since 
* `sample_fraction` is only needed as part of `TotalQty()`, 
* it is set by `SetSampleSize()`, and
* `SetSampleSize()` must be called before `TotalQty()` and only then

why not just add the code from `SetSampleSize()` to `TotalQty()` and get rid of the former and the member variable.

Reply:
Also, perhaps multiply the result of the `std::min()` by `sample_fraction_` instead of multiplying both and then taking the min?

Reply:
That's not exactly true. Yes, , `SetRequestSize()` is only called immediately before `TotalQty()` in `matl_buy_policy.cc` directly, but `TotalQty()` is also called through member functions `ReqQty()` or `NReq()`, which also happens in the same time step. If I move `sample_fraction` to be part of `TotalQty()`, then `sample_fraction` will be re-set multiple times in a single timestep, because it calls a new sample every time